test-tools/aot-analyzer/src/main.cc

```diff
@@ -0,0 +1,656 @@
+    printf("\n");
+    printf("  -data section \n");
+    for (uint32_t index = 0; index < data_section_count; index++) {
+        AOTObjectDataSection *obj_data =
+            ((AOTModule *)(aot->GetModule()))->data_sections + index;
+        printf("    -[%u] code_size:%5u Bytes    name: %s \n", index,
+               obj_data->size, obj_data->name);
+    }
+    printf("\n\n");
+}
+
+void
+DumpCompare(AoTFile *aot, WasmFile *wasm)
+{
+    const AOTTargetInfo target_info = aot->GetTargetInfo();
+    AOTModule *aot_module = (AOTModule *)(aot->GetModule());
+    WASMModuleMemConsumption aot_mem_conspn = aot->GetMemConsumption();
+
+    WASMModule *wasm_module = (WASMModule *)(wasm->GetModule());
+
+    const uint32_t aot_func_count = aot_module->func_count;
+    const uint32_t aot_code_size = aot_module->code_size;
+    struct func_info *sorted_func_ptrs = NULL;
+    sorted_func_ptrs = sort_func_ptrs(((AOTModule *)(aot->GetModule())));
+
+    const uint32_t wasm_func_count = wasm_module->function_count;
+    WASMFunction **wasm_functions = wasm_module->functions;
+
+    if (aot_func_count != wasm_func_count) {
+        printf("The number of AoT functions does not match the number of Wasm "
+               "functions. \n");
+        return;
+    }
+
+    uint32_t wasm_code_size = 0;
+    // print function Comparison Details
+    printf(
+        "|--------------------------------------------------------------------"
+        "-------------------| \n");
+    printf(
+        "|                             Function Code Size Compare             "
+        "                   | \n");
+    printf(
+        "|--------------------------------------------------------------------"
+        "-------------------| \n");
+    printf(
+        "|   ID   |  AoT Function Code Size   |  Wasm Function Code Size   |  "
+        "expansion multiple | \n");
+    printf(
+        "|--------------------------------------------------------------------"
+        "-------------------| \n");
+
+    for (uint32_t index = 0; index < aot_func_count; index++) {
+        const uint32_t aot_func_size =
+            index + 1 < aot_func_count
+                ? (uintptr_t)(sorted_func_ptrs[index + 1].ptr)
+                      - (uintptr_t)(sorted_func_ptrs[index].ptr)
+                : aot_code_size + (uintptr_t)(aot_module->code)
+                      - (uintptr_t)(sorted_func_ptrs[index].ptr);
+        const uint32_t wasm_func_size = wasm_functions[index]->code_size;
+        wasm_code_size += wasm_func_size;
+        printf(
+            "|  %4d  |    %10d Bytes       |    %10d Bytes        |  %10.2f    "
+            " "
+            "    | \n",
+            index, aot_func_size, wasm_func_size,
+            (aot_func_size * 1.0) / wasm_func_size);
+        printf(
+            "|-----------------------------------------------------------------"
+            "-"
+            "---------------------| \n");
+    }
+    printf("\n\n");
```

Comment:
Should `wasm_runtime_free(sorted_func_ptrs)`

Reply:
done